src/log/Log.cc

```diff
@@ -235,15 +235,15 @@ void Log::_flush_logbuf()
   }
 }
 
-void Log::_flush(EntryVector& t, bool requeue, bool crash)
+void Log::_flush(EntryVector& t, bool crash)
 {
   long len = 0;
+  if (t.empty()) {
```

Comment:
might need to add `assert(m_log_buf.empty())` here.

Reply:
I am afraid that this will be buggy, just in case when the log thread is stopped, then the 'm_stop == TRUE' and at the same time the 'm_new' most possibly will be empty.

```
405 void *Log::entry()             
406 {  
407   reopen_log_file();
408   {
409     std::unique_lock lock(m_queue_mutex);
410     m_queue_mutex_holder = pthread_self();
411     while (!m_stop) {
412       if (!m_new.empty()) {
413         m_queue_mutex_holder = 0;
414         lock.unlock();
415         flush();                                                                                                                                                         
416         lock.lock();
417         m_queue_mutex_holder = pthread_self();
418         continue;
419       }
420    
421       m_cond_flusher.wait(lock);
422     }
423     m_queue_mutex_holder = 0;  
424   }
425   flush();
426   return NULL;
427 }  
```

Reply:
i am not sure how this is relevant. what i suggest is
```c++
if (t.empty()) {
  assert(m_log_buf.empty());
}
```
if this is buggy, then

> In other word, m_log_buf just a buffer for the log messages in _flush() before flushing them to log files.

does not hold anymore.

Reply:
Okay, I misunderstood it.
Will fix it. Thanks.